examples/DotNetCore/EventHub/EventHubService.cs

```diff
@@ -0,0 +1,156 @@
+            string consumerGroup = _eventHubConnection.EventHubConsumerGroup;
+
+            //
+            // Check if the required configurations are available.
+            if (string.IsNullOrEmpty(eventHubConnectionString))
+            {
+                throw new ArgumentNullException(nameof(eventHubConnectionString));
+            }
+
+            if (string .IsNullOrEmpty(eventHubName))
+            {
+                throw new ArgumentNullException(nameof(eventHubName));
+            }
+
+            if (string.IsNullOrEmpty(blobStorageConnectionString))
+            {
+                throw new ArgumentNullException(nameof(blobStorageConnectionString));
+            }
+            
+            if (string.IsNullOrEmpty(blobContainerName))
+            {
+                throw new ArgumentNullException(nameof(blobContainerName));
+            }
+
+            //
+            // If no consumer group is provided, use default one.
+            if (string.IsNullOrEmpty(consumerGroup))
+            {
+                consumerGroup = EventHubConsumerClient.DefaultConsumerGroupName;
+            }
+
+            //
+            // Create blob storage client.
+            var storageClient = new BlobContainerClient(connectionString: blobStorageConnectionString,
+                                                        blobContainerName: blobContainerName);
+            
+            _processorClient = new EventProcessorClient(checkpointStore: storageClient,
+                                                        consumerGroup: consumerGroup,
+                                                        connectionString: eventHubConnectionString,
+                                                        eventHubName: eventHubName);
+        }
+
+        private async Task ProcessEventHandler(ProcessEventArgs eventArgs)
+        {
+            string rawEvent = Encoding.UTF8.GetString(eventArgs.Data.Body.ToArray());
+            IEnumerable<KeyValueEvent> events = JsonConvert.DeserializeObject<List<KeyValueEvent>>(rawEvent);
+            bool shouldRefresh = false;
+
+            foreach (var e in events)
+            {
+                var key = e.Data.Key;
+                if (IsWatchedKeyValue(key))
+                {
+                    shouldRefresh = true;
+                    break;
+                }
+            }
+
+            if (shouldRefresh)
+            {
+                await _refresher.RefreshAsync();
+                Settings = Configuration.GetSection("Demo:Settings").Get<Settings>();
+            }
+        }
+
+        private Task ProcessErrorHandler(ProcessErrorEventArgs eventArgs)
+        {
+            Console.WriteLine($"\n\nERROR: Partition: '{eventArgs.PartitionId}': an unhandled exception was encountered.\n\n");
+            Console.WriteLine(eventArgs.Exception.Message);
+            return Task.CompletedTask;
+        }
+
+        private bool IsWatchedKeyValue(string key)
```

Comment:
No need to do this.

Instead, you should register what key to watch for changes. Please use a sentinel key.